commit fcc5b94f118495b1a467edcda6c6f631691c3f69
Author: Dennis Lan <dennis.yxun@gmail.com>
Date:   Tue Jul 3 09:25:42 2012 +0800

    openjade: fix undefined Getopts error, use std namespace
    
    Using Gentoo Linux as the build host, it fails without this patch
    Use Getopt::Std in place of getopts.pl.
    
    Upstream-Status: Inappropriate [no upstream]
    Original-Author-By: Mike Gilbert <floppym@gentoo.org>
    Signed-off-by: Dennis Lan <dennis.yxun@gmail.com>

diff --git a/msggen.pl b/msggen.pl
index 0c33968..2ee3f66 100644
--- a/msggen.pl
+++ b/msggen.pl
@@ -4,6 +4,7 @@
 # See the file COPYING for copying permission.
 
 use POSIX;
+use Getopt::Std;
 
 # Package and version.
 $package = 'openjade';
@@ -18,8 +19,7 @@ $gen_c = 0;
 undef $opt_l;
 undef $opt_p;
 undef $opt_t;
-do 'getopts.pl';
-&Getopts('l:p:t:');
+getopts('l:p:t:');
 $module = $opt_l;
 $pot_file = $opt_p;
 
@@ -72,7 +72,7 @@ while (<DEF>) {
     else {
 	$field[0] =~ /^[IWQXE][0-9]$/ || &error("invalid first field");;
 	$type[$num] = substr($field[0], 0, 1);
-	$argc = int(substr($field[0], 1, 1));
+	$argc = substr($field[0], 1, 1);
     }
     $nargs[$num] = $argc;
     $field[1] =~ /^[a-zA-Z_][a-zA-Z0-9_]+$/ || &error("invalid tag");
